Include/Acidanthera/Library/OcBootManagementLib.h

```diff
@@ -530,40 +541,56 @@ typedef struct {
   // Multiple entries may share an id - allows e.g. newest version
   // of Linux install to automatically become selected default.
   //
-  CONST CHAR8    *Id;
+  CONST CHAR8              *Id;
   //
   // Entry name.
   //
-  CONST CHAR8    *Name;
+  CONST CHAR8              *Name;
   //
   // Absolute device path to file for user custom entries,
   // file path relative to device root for boot entry protocol.
   //
-  CONST CHAR8    *Path;
+  CONST CHAR8              *Path;
   //
   // Entry boot arguments.
   //
-  CONST CHAR8    *Arguments;
+  CONST CHAR8              *Arguments;
   //
   // Content flavour.
   //
-  CONST CHAR8    *Flavour;
+  CONST CHAR8              *Flavour;
   //
   // Whether this entry is auxiliary.
   //
-  BOOLEAN        Auxiliary;
+  BOOLEAN                  Auxiliary;
   //
   // Whether this entry is a tool.
   //
-  BOOLEAN        Tool;
+  BOOLEAN                  Tool;
   //
   // Whether it should be started in text mode.
   //
-  BOOLEAN        TextMode;
+  BOOLEAN                  TextMode;
   //
   // Whether we should pass the actual device path (if possible).
   //
-  BOOLEAN        RealPath;
+  BOOLEAN                  RealPath;
+  //
+  // System action. Boot Entry Protocol only. Optional.
+  //
+  OC_BOOT_SYSTEM_ACTION    SystemAction;
+  //
+  // Optional config data for system action. Boot Entry Protocol only.
+  //
+  VOID                     *ActionConfig;
```

Comment:
@vit9696 @mhaeuser - I think I have a minor issue, round here. It is reasonable for a BEP driver to want to either allocate this memory, or not. Currently, I'm doing nothing to free it, if it's allocated. What's the best pattern to follow, here? I think it's reasonably EDK-II-ish to require that this memory is always allocated, on the pool if non-NULL? (In which case, I think perhaps I should use the pre-existing LoadOptions and LoadOptionsSize to store this, and as the parameters for SYSTEM_ACTION?)